src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/IDesignerSerializationManager.cs

```diff
@@ -71,30 +71,30 @@ public interface IDesignerSerializationManager : IServiceProvider
         /// of named instances. Objects that implement IComponent will be
         /// added to the design time container if addToContainer is true.
         /// </summary>
-        object CreateInstance(Type type, ICollection arguments, string name, bool addToContainer);
+        object CreateInstance(Type type, ICollection arguments, string? name, bool addToContainer);
 
         /// <summary>
         /// Retrieves an instance of a created object of the given name, or
         /// null if that object does not exist.
         /// </summary>
-        object GetInstance(string name);
+        object? GetInstance(string name);
```

Comment:
I think name is also nullable. based on the winforms `DesignerSerializationManager`:
https://github.com/dotnet/winforms/blob/00e6b62faf223b3cf2c2417b888de71c1648fbee/src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs#L779-L782

Reply:
I don't follow... the code you linked to shows it explicitly throwing an exception if name is null.

Reply:
They stay annotated as not-null and we leave the null check in place.  Being annotated as not-null doesn't change behavior at run-time, so the null check is there to validate null isn't being passed in, just as before.